central/reports/service/v2/singleton.go

```diff
@@ -12,8 +20,77 @@ var (
 )
 
 func initialize() {
-	metadataDS := metadataDataStore.Singleton()
-	svc = New(metadataDS)
+	metadataStore := metadataDS.Singleton()
+	scheduler := initializeScheduler(metadataStore)
+	svc = New(metadataStore, scheduler)
+}
+
+func initializeScheduler(metadataDataStore metadataDS.DataStore) schedulerV2.Scheduler {
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.WorkflowAdministration)))
+
+	scheduler := schedulerV2.Singleton()
+	reportConfigDatastore := reportConfigDS.Singleton()
+
+	// Queuing pending and scheduled reports in separate routines to prevent blocking main routine during startup
+	go queuePendingReports(ctx, scheduler, metadataDataStore, reportConfigDatastore)
+	go queueScheduledReports(ctx, scheduler, reportConfigDatastore)
+
+	scheduler.Start()
+	return scheduler
+}
```

Comment:
Currently, there is no guarantee that when the scheduler is started, it is started in the correct state/snapshot. I recommend the following:

```suggestion
	scheduler := schedulerV2.Singleton()
	go schedular.Start()
	svc = New(metadataStore, scheduler)
}
```


`schedule.go`
```
func (s *scheduler) Start() {
	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
		sac.AllowFixedScopes(
			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
			sac.ResourceScopeKeys(resources.WorkflowAdministration)))

	// Queuing pending and scheduled reports in separate routines to prevent blocking main routine during startup
        s.queuePendingReports(ctx, s.metadataDataStore, s.reportConfigDatastore)
        s.queueScheduledReports(ctx, s.reportConfigDatastore)
        go s.runReports()
}
```